### PR TITLE
KV scan ensure we use batch size while iterate

### PR DESCRIPTION
**Pull request by nopcoder**
- kv dynamodb store internal scan loop parameters instead of re-calc.
- kv postgres calculate limit one and reuse by passing scan options to the iterator.
- delete unused code in db.go.

### CODE REVIEW

File: pkg/kv/postgres/store.go [hunk 1]
```diff
@@ -279,6 +280,11 @@ func (s *Store) Scan(ctx context.Context, partitionKey []byte, options kv.ScanOp
 		return nil, kv.ErrMissingPartitionKey
 	}
 
+	// normalize batch size to be the minimum between ScanPageSize and ScanOptions
+	if options.BatchSize == 0 || s.Params.ScanPageSize < options.BatchSize {
+		options.BatchSize = s.Params.ScanPageSize
+	}
```
Comment (guy-har) [inline pkg/kv/postgres/store.go:286]: 😅  saw it after my previous comment

File: pkg/kv/postgres/store.go [hunk 2]
```diff
@@ -288,20 +294,14 @@ func (s *Store) scanInternal(ctx context.Context, partitionKey []byte, options k
...
 
-	// limit set to the minimum between option's limit and the configured scan page size
-	limit := s.Params.ScanPageSize
-	if options.BatchSize > 0 && options.BatchSize < limit {
-		limit = options.BatchSize
-	}
-
 	if options.KeyStart == nil {
-		rows, err = s.Pool.Query(ctx, `SELECT partition_key,key,value FROM `+s.Params.SanitizedTableName+` WHERE partition_key=$1 ORDER BY key LIMIT $2`, partitionKey, limit)
+		rows, err = s.Pool.Query(ctx, `SELECT partition_key,key,value FROM `+s.Params.SanitizedTableName+` WHERE partition_key=$1 ORDER BY key LIMIT $2`, partitionKey, options.BatchSize)
```
Comment (guy-har) [inline pkg/kv/postgres/store.go:298]: I'd consider returning to limit, IMO it's confusing that options.BatchSize was edited

**Review by guy-har — APPROVED** [review]
LGTM
Added minor comments, 
Thanks 🙏

File: pkg/kv/dynamodb/store.go [hunk 3]
```diff
@@ -105,7 +106,7 @@ func isTableExist(ctx context.Context, svc *dynamodb.DynamoDB, table string) (bo
 		TableName: aws.String(table),
 	})
 	if err != nil {
-		if aerr, ok := err.(awserr.Error); ok && aerr.Code() == dynamodb.ErrCodeResourceNotFoundException {
+		if asErr, ok := err.(awserr.Error); ok && asErr.Code() == dynamodb.ErrCodeResourceNotFoundException {
```
Comment (N-o-Z) [inline pkg/kv/dynamodb/store.go:109]:
```suggestion
		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == dynamodb.ErrCodeResourceNotFoundException {
```

File: pkg/kv/dynamodb/store.go [hunk 4]
```diff
@@ -321,30 +322,55 @@ func (s *Store) Delete(ctx context.Context, partitionKey, key []byte) error {
...
+	// limit set to the minimum 'params.ScanLimit' and 'options.BatchSize', unless 0 (not set)
+	limit := s.params.ScanLimit
+	batchSize := int64(options.BatchSize)
+	if batchSize > 0 {
+		if limit == 0 {
+			limit = batchSize
+		} else if batchSize < limit {
+			limit = batchSize
+		}
+	}
```
[the remark this answers is not in the document]
Comment (N-o-Z) [reply pkg/kv/dynamodb/store.go:338]: preferably parenthesized for readability

**Review by N-o-Z — APPROVED** [review]
Looking good. Some suggestions

File: pkg/kv/dynamodb/store.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (nopcoder) [reply pkg/kv/dynamodb/store.go:338]: Updated as a single `if`

**Review by guy-har — APPROVED** [review]
😎

**Pull request merged**